HTTP-only build fails.

Building with `--no-default-features --features="http"` fails, but `--no-default-features --features="http ws"` succeeds.

```
$ cargo build --no-default-features --features="http"                                                                                                                                                            
   Compiling web3 v0.4.0 (file:///home/andreas/git/rust-web3)                                                                                                                                                                                 
error[E0277]: the trait bound `error::Error: std::convert::From<transports::http::url::ParseError>` is not satisfied
   --> src/transports/http.rs:116:23
    |
116 |             let url = Url::parse(url)?;
    |                       ^^^^^^^^^^^^^^^^ the trait `std::convert::From<transports::http::url::ParseError>` is not implemented for `error::Error`
    |
    = help: the following implementations were found:
              <error::Error as std::convert::From<std::io::Error>>
              <error::Error as std::convert::From<error::ErrorKind>>
              <error::Error as std::convert::From<&'a str>>
              <error::Error as std::convert::From<std::string::String>>
            and 3 others
    = note: required by `std::convert::From::from`
```

--- a/src/transports/http.rs
+++ b/src/transports/http.rs
@@ -35,6 +35,13 @@ impl From<hyper::error::UriError> for Error {
     }
 }
 
+#[cfg(all(feature = "http", not(feature = "ws")))]
+impl From<self::url::ParseError> for Error {
+    fn from(err: self::url::ParseError) -> Self {
+        ErrorKind::Transport(format!("{:?}", err)).into()
+    }
+}
+
 #[cfg(feature = "tls")]
 impl From<native_tls::Error> for Error {
     fn from(err: native_tls::Error) -> Self {

